contracts/contracts/Battle.sol

```diff
@@ -5,27 +5,222 @@ import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+  /// @dev Sets Game Item Contract to the active Game Items
+  /// @dev Script MUST add WIN_MANAGER_ROLE to THIS
+  /// @param _gameItemAddress Address of Game Item
+  constructor(address _gameItemAddress) {
+    gameItemAddress = _gameItemAddress;
+  }
+
+  ////////////////////////////////////////
+  /////////////// External ///////////////
+  ////////////////////////////////////////
+
+  /// @dev The Battle Function
+  /// @dev Checks Both the Signer and the Opponent for Items
+  /// @dev Takes the bytes32 hash and uses it with token ids to match items for a full return on both ends
+  /// @dev Emits an Event With all the data as well?
+  function battle(address opponent, bytes32 mySalt, bytes32 opponentSalt) external {
+    address _p1 = msg.sender;
+    address _p2 = opponent;
+
+    /// 1 - Confirm My Salt and Opponent Salt Are NOT Commitable i.e can battle
+    require(!_canBattle(_p1), CANNOT_BATTLE);
+    require(!_canBattle(_p2), CANNOT_BATTLE);
+
+    /// 2 - Loops through the # of Items on GameItems Contract, Attempts to Match Salt with TokenId for each player
+
+    IGameItems _gameItemsContract = _buildGameItems();
+
+    /// Loads Number of Items to Loop Through
+    uint256 _numberItems = _gameItemsContract.getNumberItems();
+    uint256 _p1TokenId = 1000; /// Initial Setting of MyTokenId
+    uint256 _p2TokenId = 1000; /// Initial Setting of Opponent Token Id
+
+    /// Loops Through # of Items
+    for (uint i = 0; i < _numberItems; i++) {
+      /// If True === Not Found yet, Continue Loop
+      if (_p1TokenId == 1000) {
+        /// Geneate Hash of Salt + "+" + tokenId
+        bytes32 _hash = keccak256(abi.encodePacked(mySalt,"+", i));
+        /// If Match -> Set Token id
+        if (_hash == committed[msg.sender]) _p1TokenId = i;
+      }
+
+      if (_p2TokenId == 1000) {
+        bytes32 _hash = keccak256(abi.encodePacked(opponentSalt,"+", i));
+        if (_hash == committed[opponent]) _p2TokenId = i;
+      }
+
+      /// If Both Players are found, break
+      if (_p1TokenId != 1000 && _p2TokenId != 1000) break;
+
+    }
+
+    require(_p1TokenId != 1000, "Player One Commitment Error");
+    require(_p2TokenId != 1000, "Player Two Commitment Error");
+
+    /// 3 - Runs Reveal and Battle Logic
+    uint8 result = _findWinner(_gameItemsContract, _p1TokenId, _p2TokenId);
+    /// 4 - Set Win/Set Lost
+    _setResult(_p1, _p2, result);
+    /// 5 - Remove Commitments for Each Player
+    _removeCommitment(msg.sender);
+    _removeCommitment(opponent);
+    /// 6 - Emit Battle Event
+    emit BattleCompleted(msg.sender, opponent, result, _p1TokenId, _p2TokenId); /// TO DO FIX
   }
 
-  // this 
-  mapping(address => mapping(uint => PlayedItem)) public rounds;
+  /// @dev Enables Player to Commit an Item
+  /// @param commitment Takes a hashed commit and stores on chain
+  function commitItem(bytes32 commitment, uint256 tokenId) external {
```

Comment:
calling this function means that tokenId is public

Reply:
No worries, will remove. Thanks for clarifying; I was unsure on that front.